### PR TITLE
Remove unnecessary CDN in a Box waits

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

* [x] This PR is not related to any Issue 
* This PR makes CDN in a Box start up faster.
    - Lets you skip things waiting for the enroller initial data load to complete. This saves about 24 seconds.
    - Lets you set `DIG_IP_RETRY` from `set-to-ips-from-dns.sh` to any value, letting you optionally skip this step . When set to 0, this saves 30 seconds per missing service. So, in the case of the Traffic Ops API tests, where the `trafficportal`, `trafficmonitor`, and `trafficrouter` are not included, this saves 90 seconds.
    - Traffic Vault now waits to enroll itself until after Riak finishes starting up. Saves about 24 seconds (the amount of time the enroller initial data load takes).
 
 Being able to skip the enroller initial data load is a bugfix for the Traffic Ops API tests. Right now, if you start the `integration` service (from `docker-compose.traffic-ops-test.yml`) when starting all of the other CDN in a Box services, the Traffic Ops API v1 tests fail (but v2 and v3 succeed) at the end of `TestCacheGroupsDeliveryServices` because the enroller continues seeding data after `Teardown()` truncates all the tables.
```
    types_test.go:96: ---- DeleteTestTypes ----
    types_test.go:108: cannot DELETE Type by name: 400 Bad Request[400] - Error requesting Traffic Ops https://trafficops.infra.ciab.test:443/api/1.3/types/135 {"alerts":[{"text":"cannot delete type because it is being used by a cachegroup","level":"error"}]}
         - {[]}
    types_test.go:117: expected Type name: ORG_LOC to be deleted
    types_test.go:108: cannot DELETE Type by name: 400 Bad Request[400] - Error requesting Traffic Ops https://trafficops.infra.ciab.test:443/api/1.3/types/136 {"alerts":[{"text":"cannot delete type because it is being used by a cachegroup","level":"error"}]}
         - {[]}
    types_test.go:117: expected Type name: MID_LOC to be deleted
    types_test.go:108: cannot DELETE Type by name: 400 Bad Request[400] - Error requesting Traffic Ops https://trafficops.infra.ciab.test:443/api/1.3/types/145 {"alerts":[{"text":"cannot delete type because it is being used by a cachegroup","level":"error"}]}
         - {[]}
    types_test.go:117: expected Type name: EDGE_LOC to be deleted
--- FAIL: TestCacheGroupsDeliveryServices (8.69s)
```

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Documentation
- CI tests

## What is the best way to verify this PR?

* Make sure the Traffic Ops API tests pass when `dig` retries and enroller initial data loading is disabled:
    ```
    export DIG_IP_RETRY=0 LOAD_TRAFFIC_OPS_DATA=false;
    docker-compose up --no-deps -d edge dns db smtp trafficops-perl trafficops trafficvault;
    docker-compose -f docker-compose.traffic-ops-test.yml -f docker-compose.yml up integration;
    ```

* Make sure CDN in a Box still works when started normally.

## If this is a bug fix, what versions of Traffic Control are affected?

For the Traffic Ops API tests:
 
- master (f6a8056513)
- 5.0.0 (RC1)

## The following criteria are ALL met by this PR

- [x] This PR fixes tests
- [x] This PR includes documentation
- [x] An update to CHANGELOG.md is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: infrastructure/cdn-in-a-box/variables.env [hunk 1]
```diff
@@ -122,3 +122,5 @@ AUTO_SNAPQUEUE_ENABLED=true
 AUTO_SNAPQUEUE_SERVERS=trafficops,trafficops-perl,trafficmonitor,trafficrouter,trafficvault,edge,mid-01,mid-02
 AUTO_SNAPQUEUE_POLL_INTERVAL=2
 AUTO_SNAPQUEUE_ACTION_WAIT=2
+LOAD_TRAFFIC_OPS_DATA=${LOAD_TRAFFIC_OPS_DATA:-true}
```
Comment (rawlinp) [inline infrastructure/cdn-in-a-box/variables.env:125]: These don't appear to be evaluating as if this were a bash script. This variable is getting set to literal `'${LOAD_TRAFFIC_OPS_DATA:-true}'`

**Review by rawlinp — CHANGES_REQUESTED** [review]

File: infrastructure/cdn-in-a-box/variables.env (hunk 1 above)
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/variables.env:125]: Where are you seeing `LOAD_TRAFFIC_OPS_DATA` being set to literal `${LOAD_TRAFFIC_OPS_DATA:-true}`? I tested on macOS with Docker Desktop and on Linux and was unable to reproduce. Testcase:

```shell
docker-compose run --rm edge sh -c 'echo $LOAD_TRAFFIC_OPS_DATA'
```
echos `true` for me.

Comment (rawlinp) [reply infrastructure/cdn-in-a-box/variables.env:125]: I added an echo statement to the trafficops-init.sh script: 
```
echo "LOAD_TRAFFIC_OPS_DATA = $LOAD_TRAFFIC_OPS_DATA. Skipping loading of traffic ops data."
```
output:
```
+ echo 'LOAD_TRAFFIC_OPS_DATA = ${LOAD_TRAFFIC_OPS_DATA:-true}. Skipping loading of traffic ops data.'
LOAD_TRAFFIC_OPS_DATA = ${LOAD_TRAFFIC_OPS_DATA:-true}. Skipping loading of traffic ops data.
```

When I run that command you pasted, I don't see any output. Maybe a difference in docker-compose versions? 
```
$ docker-compose version
docker-compose version 1.18.0, build 8dd22a9
docker-py version: 2.6.1
CPython version: 3.6.8
OpenSSL version: OpenSSL 1.0.2k-fips  26 Jan 2017

$ docker version
Client:
 Version:           18.06.3-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        d7080c1
 Built:             Wed Feb 20 02:26:51 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.3-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       d7080c1
  Built:            Wed Feb 20 02:28:17 2019
  OS/Arch:          linux/amd64
  Experimental:     false

$ cat /etc/centos-release
CentOS Linux release 7.4.1708 (Core)
```

Comment (rawlinp) [reply infrastructure/cdn-in-a-box/variables.env:125]: Looking at https://docs.docker.com/compose/env-file/ it doesn't seem to indicate that variable interpolation can be done in a .env file. It seems like it only accepts `VAR=VAL`. Maybe you have that variable set in your environment already?

Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/variables.env]: From my testing, parameter expansion only works in .env files in docker/compose@6d2658ea65 and up (stable docker-compose versions 1.26.0 and up), when docker-compose started using `python-dotenv` to process .env files.

Specifying those environment variables in `docker-compose.yml` in 4b7d9a0a4e.

**Review by rawlinp — APPROVED** [review]
Works as expected now in my old version of docker-compose

**Pull request merged**